DataScience/Experimentation.py

```diff
@@ -406,24 +420,32 @@ def main(args):
             best_command = results[0]
             best_commands.append(['Hyper2', results[0]])
 
-        # TODO: Repeat above process of tuning parameters and interactions until convergence / no more improvements.
-
     t1 = datetime.now()
     print("\n\n*************************")
     Logger.info("Best parameters found after {}:".format((t1-t0)-timedelta(microseconds=(t1-t0).microseconds)))
     best_command.prints()
     print("*************************")
 
+    # TODO: Repeat above process of tuning parameters and interactions until convergence / no more improvements.
+
+    t2 = datetime.now()
     if args.generate_predictions:
         _ = generate_predictions_files(args.file_path, best_commands, args.log_type)
         t2 = datetime.now()
         Logger.info('Predictions Generation Time: {}'.format((t2-t1)-timedelta(microseconds=(t2-t1).microseconds)))
-        Logger.info('Total Elapsed Time: {}'.format((t2-t0)-timedelta(microseconds=(t2-t0).microseconds)))
-        
+
+    t3 = datetime.now()
```

Comment:
Maybe you want this to be t2?

Reply:
See comment from earlier about t2. t2 should stand for the time stamp when generate_predictons was completed and t3 when generate_models was completed. If those options are not even requested, then also the code should work. So I initialize them before the conditional check and reset them to the end time if the step does execute.